core/src/parser/mod.rs

```diff
@@ -0,0 +1,107 @@
+use crate::parser::helpers::{flatten_items, has_typeshare_annotation, TYPESHARE};
+
+use crate::rust_types::{RustEnum, RustItem, RustStruct, RustTypeAlias};
+
+use syn::Item;
+
+mod enum_parser;
+mod error;
+/// Helpers used throughout the parsing procedure.
+pub mod helpers;
+mod struct_parser;
+mod type_alias_parser;
+pub use error::ParseError;
+
+/// Trait used to parse the input file and provide our internal representations for various Rust constructs.
+pub trait Parser<'a> {
```

Comment:
I'd like to remove this trait and just use inherent methods on the relevant types. I'm not seeing a need for the trait, since there isn't any functionality that's generic over some `T: Parser`.

Reply:
Actually, because this refactoring includes implementation changes rather than a verbatim copy of the old items into the new submodules, I'd like to ask that you revert these changes here and re-submit them as a new PR so that they can be more specifically reviewed.